src/main/java/net/dv8tion/jda/core/requests/restaction/order/CategoryOrderAction.java

```diff
@@ -0,0 +1,152 @@
+/*
+ *     Copyright 2015-2017 Austin Keener & Michael Ritter & Florian Spieß
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.core.requests.restaction.order;
+
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.*;
+import net.dv8tion.jda.core.exceptions.InsufficientPermissionException;
+import net.dv8tion.jda.core.requests.Route;
+import net.dv8tion.jda.core.utils.Checks;
+import okhttp3.RequestBody;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import java.util.List;
+
+/**
+ * Implementation of {@link net.dv8tion.jda.core.requests.restaction.order.OrderAction OrderAction}
+ * similar in functionality to {@link net.dv8tion.jda.core.requests.restaction.order.ChannelOrderAction ChannelOrderAction},
+ * but constrained to the bounds of a single {@link net.dv8tion.jda.core.entities.Category Category}.
+ * <br>To apply the changes you must finish the {@link net.dv8tion.jda.core.requests.RestAction RestAction}.
+ *
+ * <p>Before you can use any of the {@code move} methods
+ * you must use either {@link #selectPosition(Object) selectPosition(Channel)} or {@link #selectPosition(int)}!
+ *
+ * @param  <T>
+ *         The type of {@link net.dv8tion.jda.core.entities.Channel Channel} to move
+ *         using this CategoryOrderAction, either {@link net.dv8tion.jda.core.entities.TextChannel TextChannel},
+ *         or {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel}.
+ *
+ * @author Kaidan Gustave
+ */
+public class CategoryOrderAction<T extends Channel> extends OrderAction<T, CategoryOrderAction<T>>
```

Comment:
What is the reason not to just extend on ChannelOrderAction here?

Reply:
Agreed

Reply:
If I remember correctly it has to do with the fact ChannelOrderAction's constructor populates the orderList with all the channels from the Guild argument, meaning it wouldn't work.
A solution might be to add a package private default constructor to ChannelOrderAction, then call that when extending it as CategoryOrderAction's super class, but I wasn't sure if that would be appropriate so I didn't do it here.

Reply:
I am going to leave this as it is currently.
If you would like me to make changes to have it extend `ChannelOrderAction` just say the word and I will, however I personally don't see a reason. There are arguable documentation inconsistencies between the two which could lead to confusion, and I believe future changes to either could lead to issues with `CategoryOrderAction`.

Reply:
```java
public ChannelOrderAction(Guild guild, ChannelType type, Collection<T> channels) 
{
    super(guild.getJDA(), Route.Guilds.MODIFY_CHANNELS.compile(guild.getId()));
    this.guild = guild;
    this.type = type;
    this.orderList.addAll(chans);
}
```